### PR TITLE
Fix Wukong activity snapshot tool field mapping

### PR DESCRIPTION
**Pull request by Snssn**
## Summary
- Map Wukong `FILE_READ`, `SEARCH`, `DIRECTORY_LIST`, `SKILL`, and `ARTIFACT` activity snapshots to canonical tool arguments/results.
- Preserve `FILE_WRITE.file_path` as canonical tool call arguments.
- Add Wukong unit coverage for activity snapshot payload shapes observed from the API.

## Test plan
- [x] `npm test -- tests/unit/inputs/wukong-input.test.ts`
- [x] `npm run typecheck --silent`

### CODE REVIEW

**Review by ralf0131 — APPROVED** [review]
### Summary
Extends the Wukong input handler to map additional activity types (FILE_READ, SEARCH, SKILL, ARTIFACT) to canonical tool fields, and enhances existing cases (FILE_WRITE, DIRECTORY_LIST) with fallback field naming. The new `compactObject` utility cleanly filters null/undefined values, reducing boilerplate across all new cases.

### Findings
No issues found. Code is well-structured and backward-compatible — existing cases retain their behavior with added fallbacks, and new cases are purely additive.

### Tests
Comprehensive test coverage for all new activity types, including edge cases (ARTIFACT with undefined args, FILE_WRITE with `file_path` fallback).

---
*Automated review by github-manager-bot*

File: src/inputs/wukong/wukong-input.ts [hunk 1]
```diff
@@ -1221,17 +1223,48 @@ export class WukongInput extends BaseInput {
           args = content.command ? { command: content.command } : undefined;
           result = { output: content.output, exit_code: content.exit_code };
           break;
-        case 'FILE_WRITE':
-          args = content.path ? { path: content.path } : undefined;
+        case 'FILE_WRITE': {
+          const filePath = content.path ?? content.file_path;
+          args = filePath !== undefined ? { path: filePath } : undefined;
```
Comment (linrunqi08) [inline src/inputs/wukong/wukong-input.ts]: **[Medium]** `FILE_WRITE` 使用 `filePath !== undefined`，当 `path`/`file_path` 均为 `null` 时会产出 `{ path: null }`

**影响:** canonical tool arguments 携带非法 path；旧逻辑对 falsy path 会省略 arguments
**建议:** 改为 `compactObject({ path: content.path ?? content.file_path })`，或 `filePath != null`（同时拒绝 null）

File: src/inputs/wukong/wukong-input.ts [hunk 2]
```diff
@@ -1221,17 +1223,48 @@ export class WukongInput extends BaseInput {
...
           result = { status: content.status ?? 'done' };
           break;
+        }
+        case 'FILE_READ':
+          args = compactObject({ path: content.path, start_line: content.start_line });
+          result = compactObject({
+            content: content.content,
+            snippet: content.snippet,
+            total_lines: content.total_lines,
+            status: content.status,
```
Comment (linrunqi08) [inline src/inputs/wukong/wukong-input.ts:1236]: **[Medium]** 本 PR 将 `FILE_READ`/`SKILL` 的 `status`/`error_message` 写入 result，但同函数内 `tool.result.status`（约 L1297）仍仅由 `exit_code` 决定

**影响:** 失败快照会出现顶层 `success` 与 result 内错误字段并存，下游按 canonical status 统计会漏失败（hasError 逻辑为既有；矛盾被本 PR 放大）
**建议:** 对非 TERMINAL 扩展失败判定（`status ∈ {error,failed,failure}` 或非空 `error_message`），并按 schema 设置 `error.*`；或明确用测试锁定“仅透传、不改顶层 status”的产品意图

**Comment by linrunqi08** [comment]
## 🔍 Code Review Summary

| Severity | Count |
|----------|-------|
| Critical | 0 |
| High     | 0 |
| Medium   | 2 |
| Low      | 2 |

### 总体结论

发现 4 个非阻断问题（2 Medium + 2 Low），可合入但建议后续改进。

---
*评审报告详见: `code-review/pr-139/final-report.md`*  
*Generated by LoongSuite-Pilot Code Review Agent*

**Review by ralf0131 — APPROVED** [review]
### Summary
Re-review of new commit `dd1f6f8` (since previous approval on `e7fb994`).

The new commit refines error/status handling: it introduces `resolveActivityResultStatus()` for consistent multi-signal failure detection (exit_code, error_message, status string) and adds an `error.message` attribute when an error message is present. The `FILE_WRITE` case is simplified to use `compactObject` directly.

No issues found. `compactObject` correctly strips null/undefined without over-filtering falsy values like `0`. Test coverage is comprehensive — both the happy-path field mapping and the failure-status/null-argument edge cases are covered.

---
*Automated review by github-manager-bot*

File: src/inputs/wukong/wukong-input.ts (hunk 1 above)
Comment (linrunqi08) [reply src/inputs/wukong/wukong-input.ts]: ✅ **已修复 (verified @ dd1f6f8)** — `FILE_WRITE` 现改为 `compactObject({ path: content.path ?? content.file_path })`，`compactObject` 过滤 `null/undefined`，`file_path: null` 时 arguments 整体省略。测试 `wukong-input.test.ts` 已断言此场景 arguments 为 `undefined`。

*Generated by LoongSuite-Pilot Code Review Agent*

File: src/inputs/wukong/wukong-input.ts (hunk 2 above)
Comment (linrunqi08) [reply src/inputs/wukong/wukong-input.ts:1236]: ✅ **已修复 (verified @ dd1f6f8)** — 新增 `resolveActivityResultStatus(content)`：`exit_code!=0` / 非空 `error_message` / `status∈{error,failed,failure}` → `failure`，`cancelled/canceled` → `cancelled`，并透传 `error.message`。`failure` 经归一化自动补 `error.type=_OTHER`。测试覆盖 FILE_READ/SKILL 失败 → `tool.result.status=failure` + `error.message`。顶层 status 与 result 内错误字段不再矛盾。

*Generated by LoongSuite-Pilot Code Review Agent*

**Comment by linrunqi08** [comment]
## 🔍 Code Review Summary (Round 2 — 增量复核)

复核对象：head `dd1f6f8`（新增 commit `fix: handle Wukong activity errors consistently`）

| Severity | Count (本轮新增) |
|----------|------------------|
| Critical | 0 |
| High     | 0 |
| Medium   | 0 |
| Low      | 0 |

### 上轮 Findings 复核

| # | 严重度 | 状态 |
|---|--------|------|
| M1 `FILE_WRITE` null path 产出 `{path:null}` | Medium | ✅ Fixed |
| M2 失败态未反映到 `tool.result.status` | Medium | ✅ Fixed |
| L1 `FILE_READ` 未兼容 `file_path` | Low | ⚠️ 仍未处理（推测性、非阻断） |
| L2 测试未锁失败语义/旧字段 | Low | ✅ Fixed |

### Lifecycle Verdict

| Check | Result |
|-------|--------|
| 资源释放 | PASS |
| 死锁/卡死风险 | PASS |
| 状态恢复正确性 | PASS |

### 总体结论

第二个 commit 精准修复了上一轮两条 Medium：`FILE_WRITE` 用 `compactObject` 拒绝 `null` path；新增 `resolveActivityResultStatus` 将 `error_message`/`status` 语义反映到顶层 `tool.result.status` 并透传 `error.message`（`failure` 自动补 `error.type=_OTHER`，`cancelled` 为合法枚举）。失败语义已由新单测锁定，CI (Node 18/20/22) 全绿。**未发现新增阻断问题，可以合入。**

唯一残余为 Low L1（`FILE_READ` 兼容 `file_path`），推测性、可后续跟进，不阻断本 PR。

### Highlights
- 修复最小化、仅动 Input 映射层，回归面可控。
- 失败判定抽为纯函数 `resolveActivityResultStatus`，可读可测。
- 新增失败态单测把两条 Medium 用测试固定，防回归。
- 复用既有 canonical 归一化链，无重复造轮子。

---
*评审报告详见: `code-review/pr-139/final-report-round2.md`*
*Generated by LoongSuite-Pilot Code Review Agent*

**Comment by ralf0131** [comment]
**Issue Evaluation**

Category: `enhancement` | Status: **Already Implemented**

The Wukong activity snapshot tool field mapping described in this issue is already present in `src/inputs/wukong/wukong-input.ts`:
- `FILE_READ`, `SEARCH`, `DIRECTORY_LIST`, `SKILL`, `ARTIFACT` activity snapshots mapped to canonical tool arguments/results ✓
- `FILE_WRITE.file_path` preserved as canonical tool call arguments ✓
- Activity snapshot payload shapes handled for all observed API patterns ✓

No further action needed. This issue can be closed as completed.

---
*Automated evaluation by github-manager*
